### PR TITLE
Add ettus USRP support (gr-uhd) in GNUradio

### PR DESCRIPTION
**Pull request by repk**
This first adds uhd package for Ettus USRP SDR devices, as well as enabling UHD support in gnuradio.
This also adds two GNUradio packages for FM RDS and HDR support. And finally this enables Qt support in GNUradio as gr-nrsc5 examples use it.

In order to have Qt support in GNUradio, qwt should be compiled with Qt4, so I added a qwt-qt4 package. But maybe current qwt package should be renamed to qwt-qt5 instead ?

### CODE REVIEW

**Comment by Hoshpak** [comment]
Please don't add any packages that depend on qt4 any more. We are in the process of dropping qt4 support and will remove packages depending on it and finally qt4 itself from the repository in the future.

**Comment by repk** [comment]
I wouldn't have if GNUradio current release version did support Qt5. But it doesn't. Would you rather like to go from current stable release to a git commit for GNUradio instead ? On the other hand ubuntu seems to have patched gnuradio to use Qt5 maybe we should do the same ?

**Comment by repk** [comment]
So current status is that I used patches from https://sources.debian.org/src/gnuradio/3.7.13.4-2/debian/patches/. So now Qt5 is used instead of Qt4.

Because uhd and gnuradio build times exceed travis' limit, I have added "[ci skip]" in commit messages, but because travis uses the whole merge commit, those are still built. Now that is the only reason of travis failure.

I have locally succeed to build this patchset for x86_64-musl also but cannot cross-build this for aarch64 nor arm because of nocross option in gnuradio and python-numpy.

**Comment by repk** [comment]
I renamed gr- packages into gnuradio- and fix the SPDX identifier.

On a side note I also test aarch64 compilation (cross compilation is deactivated for gnuradio) by using binfmt + qemu-user:
```
./xbps-src -m masterdir-aarch64-musl binary-boostrap aarch64-musl
cp /usr/bin/qemu-aarch64-static masterdir-aarch64-musl/usr/bin
./xbps-src -m masterdir-aarch64-musl pkg ...
```
Package creation went well.

**Comment by the-maldridge** [comment]
@Hoshpak any opinions on the patches?

**Comment by Hoshpak** [comment]
I don't have a strong opinion about them and no time to completely review them right now. I briefly looked at them and they seem fine, the fact that they have been vetted by Debian and Ubuntu that usually would be good enough for me.

File: srcpkgs/gnuradio-nrsc5/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+gnuradio-nrsc5-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/lib/cmake
+	}
+}
+
```
Comment (maxice8) [inline srcpkgs/gnuradio-nrsc5/template]: extra blank line, ,remove.

File: srcpkgs/uhd/template [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+uhd-doc_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
+	short_desc+=" - doc files"
+	pkg_install() {
+		vmove usr/share/doc
+	}
+}
+
+uhd-progs_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
```
Comment (maxice8) [inline srcpkgs/uhd/template]: no need for adding makedepends, that is just for -devel packages.

File: srcpkgs/uhd/template [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/share/man/man1
+		vmove usr/lib/pkgconfig
+	}
+}
+
+uhd-doc_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
```
Comment (maxice8) [inline srcpkgs/uhd/template]: should have no dependencies and be noarch since documentation is not machine code.

File: srcpkgs/uhd/template [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+wrksrc=${pkgname}-${version}
+build_wrksrc=host
+
+uhd-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/share/man/man1
+		vmove usr/lib/pkgconfig
```
Comment (maxice8) [inline srcpkgs/uhd/template]: should include the *.so symlink

File: srcpkgs/gnuradio-nrsc5/template [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'gnuradio-nrsc5'
+pkgname=gnuradio-nrsc5
+version=20180917
+revision=1
+_githash=e3f06d9acc0f3cc915b2296c717476ad5b045b05
```
Comment (maxice8) [inline srcpkgs/gnuradio-nrsc5/template]: please ask upstream to tag a release~

File: srcpkgs/gnuradio/template [hunk 6]
```diff
@@ -10,13 +10,15 @@ pycompile_module="gnuradio grc_gnuradio pmt"
...
-depends="pygtk python-cheetah python-lxml python-numpy"
+ libusb-devel pygtk-devel volk-devel wxPython-devel uhd-devel python-devel
+ python-PyQt5 qt5-devel qwt-devel"
+depends="pygtk python-cheetah python-lxml python-numpy python-PyQt5"
 short_desc="Framework for software defined radio"
 maintainer="Andrew Benson <abenson+void@gmail.com>"
 license="GPL-3.0-or-later"
 homepage="https://www.gnuradio.org"
 distfiles="https://github.com/gnuradio/gnuradio/archive/v${version}.tar.gz"
+patch_args="-Np1"
```
Comment (maxice8) [inline srcpkgs/gnuradio/template]: put below checksum=

File: srcpkgs/gnuradio-nrsc5/template (hunk 5 above)
Comment (repk) [reply srcpkgs/gnuradio-nrsc5/template]: asked https://github.com/argilo/gr-nrsc5/issues/9.

File: srcpkgs/uhd/template [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+uhd-doc_package() {
+	short_desc+=" - doc files"
+	noarch=yes
+	pkg_install() {
+		vmove usr/share/doc
+	}
+}
+
+uhd-progs_package() {
```
Comment (maxice8) [inline srcpkgs/uhd/template]: progs is reserved for programs meant to be used by the end-user.

I'd just move the examples to the -devel package and remove the tests

File: srcpkgs/gnuradio-rds/template [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+license="GPL-2.0-or-later"
+homepage="https://github.com/bastibl/gr-rds/"
+distfiles="https://github.com/bastibl/gr-rds/archive/v${version}.tar.gz"
+checksum=503574ed44e00caba1fb2aed7722fac487c318e57cb438d3af7833cd91dffb22
+
+gnuradio-rds-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
```
Comment (maxice8) [inline srcpkgs/gnuradio-rds/template]: missing:

`vmove "usr/lib/*.so"`

File: srcpkgs/gnuradio-rds/template [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+	}
+}
```
Comment (maxice8) [inline srcpkgs/gnuradio-rds/template]: Problably also needs a package for its python bindings

File: srcpkgs/gnuradio-nrsc5/template [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+maintainer="Remi Pommarel <repk@triplefau.lt>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/argilo/gr-nrsc5/"
+distfiles="https://github.com/argilo/gr-nrsc5/archive/${_githash}.tar.gz"
+checksum=d0d0ec9dc5bcdf6864f99a9b89e7f1eff3ecd6ec1748a4c5c9647bd0762d9913
+
+gnuradio-nrsc5-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
```
Comment (maxice8) [inline srcpkgs/gnuradio-nrsc5/template]: missing

`vmove "usr/lib/*.so"`

File: srcpkgs/uhd/template [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+uhd-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/share/man/man1
+		vmove usr/lib/pkgconfig
+		vmove usr/lib/uhd/examples
+		vmove "usr/lib/*.so"
```
Comment (maxice8) [inline srcpkgs/uhd/template]: missing

`vmove usr/lib/cmake`

File: srcpkgs/uhd/template [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+checksum=e18d0524cbf571be4847fd7f971dc30c37efd9e7a333761b74e1266a07cbd35b
+wrksrc=${pkgname}-${version}
+build_wrksrc=host
+
+post_install() {
+	rm -rf ${DESTDIR}/usr/lib/uhd/tests
+}
+
+uhd-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
```
Comment (maxice8) [inline srcpkgs/uhd/template]: only dependency arrcoding to pkgconfig file is gnuradio-devel

Comment (repk) [reply srcpkgs/uhd/template]: Are you sure ? I mean this package is fine without gnuradio-devel. It's gnuradio that needs it not the otherway around.

Comment (maxice8) [reply srcpkgs/uhd/template]: Yes, xpcdeps seems to be borked in that regard.

**Comment by repk** [comment]
Should I keep rebasing that or should I consider this as a [nomerge] ?

**Comment by Vaelatern** [comment]
@maxice8 what do you think?

**Comment by maxice8** [comment]
On Mon, Jan 14, 2019 at 08:44:52AM -0800, Toyam Cox wrote:
> @maxice8 what do you think?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/4563#issuecomment-454074076

This PR is super big, i'd rather it be broken up

**Pull request closed without merging**